lib/puppet/configurer.rb

```diff
@@ -425,6 +432,18 @@ def run_internal(options)
   end
   private :run_internal
 
+  def valid_server_environment?
+    session = Puppet.lookup(:http_session)
+    begin
+      fs = session.route_to(:fileserver)
+      response = fs.get_file_metadatas(path: URI(Puppet[:pluginsource]).path, recurse: :false, environment: @environment)
+      return response[0].success?
+    rescue => detail
+      Puppet.notice _("Environment '%{environment}' not found on server. Skipping initial pluginsync. Error detail: %{detail}") % {environment: @environment, detail: detail}
+      return false
```

Comment:
Wondering if we should print a different message/level if the request fails because "the environment doesn't exist" vs "everything else - couldn't connect to the server, SSL problem, timeout, etc"? Maybe just print `Puppet.notice` for the first, but call `Puppet.log_exception(detail, detail.message)` in the second case?